services/pull/check.go

```diff
@@ -81,13 +81,6 @@ func CheckPullMergable(ctx context.Context, doer *user_model.User, perm *models.
 		return nil
 	}
 
-	if pr.ProtectedBranch != nil && (!pr.ProtectedBranch.HasEnoughApprovals(pr) ||
-		pr.ProtectedBranch.MergeBlockedByRejectedReview(pr) ||
-		pr.ProtectedBranch.MergeBlockedByOfficialReviewRequests(pr) ||
-		pr.ProtectedBranch.MergeBlockedByOutdatedBranch(pr)) {
-		return ErrUserNotAllowedToMerge
-	}
```

Comment:
What happened to this code block?

Reply:
checked inside of CheckPullBranchProtections that just was lost since err was overwritten

see https://github.com/go-gitea/gitea/pull/19574 for the pure bugfix :)